ccm_web/server/src/api/api.service.ts

```diff
@@ -40,16 +40,20 @@ export class APIService {
   async getCourseSections (userLoginId: string, courseId: number): Promise<CanvasCourseSections | APIErrorData> {
     const requestor = await this.canvasService.createRequestorForUser(userLoginId, '/api/v1/')
     try {
-      const endpoint = `courses/${courseId}`
-      const queryParams = {'include': ['sections', 'total_students']}
-      logger.debug(`Sending request to Canvas - Endpoint: ${endpoint}; Method: GET`)
-      const response = await requestor.get<CanvasCourse>(endpoint, queryParams)
-      logger.debug(`Received response with status code ${response.statusCode}`)
-      const course = response.body
-      return {
-        id: course.id, name: course.name,
-        sections: course.sections,
-        total_students: course.total_students}
+      const courseEndpoint = `courses/${courseId}`
+      logger.debug(`Sending request to Canvas - Endpoint: ${courseEndpoint}; Method: GET`)
+      const courseResponse = await requestor.get<CanvasCourse>(courseEndpoint)
+      logger.debug(`Received response with status code ${courseResponse.statusCode}`)
+      const course = courseResponse.body
+
+      const sectionsEndpoint = `courses/${courseId}/sections`
+      const queryParams = {'include': ['total_students']}
+      logger.debug(`Sending request to Canvas - Endpoint: ${sectionsEndpoint}; Method: GET`)
+      const sectionsResponse = await requestor.get<object>(sectionsEndpoint, queryParams)
+      logger.debug(`Received response with status code ${sectionsResponse.statusCode}`)
+      const courseSections = sectionsResponse.body
```

Comment:
This API itself is good enough and don't need the `courses/${courseId}`

Reply:
OK.  If you think so.  I thought the course name was also required.

If this is all that's needed, that will simplify things a little.

Reply:
No, Section name and id, student count in each section is Good. 

Ideally we want all the users (teachers, students, TA etc) count in the section roster, but that info is not something easily available with this API `/course/:courseID/sections` with out making extra enrollment API call which we want to avoid.